### PR TITLE
Add MavenArtifactFinder to find a plugin Maven artifact from a local Maven repository.

### PR DESCRIPTION
**Pull request by dmikurube**
@muga Finally, I've split this part from #684. Itself is ready to review and merge -- can you have a look?

### CODE REVIEW

File: embulk-core/src/main/java/org/embulk/plugin/maven/MavenArtifactFinder.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        final Path absolutePath;
+        try {
+            absolutePath = localMavenRepositoryPath.normalize().toAbsolutePath();
+        }
+        catch (IOError ex) {
+            throw new MavenRepositoryNotFoundException(localMavenRepositoryPath, ex);
+        }
+        catch (SecurityException ex) {
+            throw new MavenRepositoryNotFoundException(localMavenRepositoryPath, ex);
+        }
```
Comment (muga) [inline embulk-core/src/main/java/org/embulk/plugin/maven/MavenArtifactFinder.java]: minor: Cannot we catch handle both of `IOExrror` and `SecurityException` in a single catch statement? like `catch (IOError | SecurityException) {`

Comment (dmikurube) [inline embulk-core/src/main/java/org/embulk/plugin/maven/MavenArtifactFinder.java]: @muga Ah, at first I thought to change their Exception messages while they're the same at the moment.

I'm keeping this as-is for the time being, and later considering changing messages.

Comment (muga) [inline embulk-core/src/main/java/org/embulk/plugin/maven/MavenArtifactFinder.java]: Ok, understand.

**Comment by dmikurube** [comment]
Thanks! Merging.

**Pull request merged**